com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -366,21 +392,56 @@ protected NetworkBehaviour GetNetworkBehaviour(ushort behaviourId)
 
         internal void InternalOnNetworkSpawn()
         {
+            IsSpawned = true;
             InitializeVariables();
+            UpdatePropertyStates();
+            UpdateOwnership();
+        }
+
+        internal void InternalOnNetworkDespawn()
+        {
+            IsSpawned = false;
+            UpdatePropertyStates();
+            UpdateOwnership();
         }
 
-        internal void InternalOnNetworkDespawn() { }
+        internal void UpdateOwnership()
+        {
+            if (NetworkObject != null)
+            {
+                IsOwnedByServer = NetworkObject.IsOwnedByServer;
+                IsOwner = NetworkObject.IsOwner;
+                OwnerClientId = NetworkObject.OwnerClientId;
+            }
+            else // Shouldn't happen, but if it does then just clear out everything
+            {
+                IsOwnedByServer = IsOwner = false;
+                OwnerClientId = 0;
+            }
+        }
```

Comment:
move this logic into `UpdateNetworkProperties()` and maintain updating all net props in one place. easier to maintain.